lib/ansible/plugins/httpapi/dcnm.py

```diff
@@ -61,28 +61,33 @@ def send_request(self, method, path, json=None):
             path = str(path)
             if path[0] != '/':
                 msg = 'Value of <path> does not appear to be formated properly'
-                raise Exception(self._return_error(None, method, path, msg))
-            response, response_data = self.connection.send(path, json, method=method, headers=self.headers, force_basic_auth=True)
-            self._verify_response(response, method, path)
-            response_value = self._get_response_value(response_data)
-            return self._response_to_json(response_value)
+                raise Exception(self._return_info(None, method, path, msg))
+            response, rdata = self.connection.send(path, json, method=method,
+                                                   headers=self.headers,
+                                                   force_basic_auth=True)
+            return self._verify_response(response, method, path, rdata)
         except Exception as e:
             eargs = e.args[0]
             if isinstance(eargs, dict) and eargs.get('METHOD'):
                 return [eargs]
```

Comment:
This is where the failures are getting enclosed in list.

Reply:
Good catch.  I made the change.  Here is a failure message as reported by Ansible with this change.

```console
   "msg": {
        "DATA": [
            {
                "column": 0,
                "entity": "n9k-109:Loopback90",
                "line": 0,
                "message": "SET of IP/Subnet resource is failed for Input: Fabric=wiebe55, Pool=LOOPBACK0_IP_POOL, Entity Type=DeviceInterface, Entity=FOX1821H035~loopback90, Resource=10.10.10.89. Reason: 10.10.10.89 is allocated to Switch= [FOX1821H035/n9k-109]",
                "reportItemType": "ERROR"
            }
        ],
        "MESSAGE": "Internal Server Error",
        "METHOD": "POST",
        "REQUEST_PATH": "https://10.122.197.6:443/rest/interface",
        "RETURN_CODE": 500
    }
```

Reply:
Good catch.  I made the change.  Here is the data we get back on failure now.

```python
{
    'DATA': [{'column': 0,
           'entity': 'n9k-109:Loopback90',
           'line': 0,
           'message': 'Loopback90 is already in use.  You can modify the '
                      'config by editing the interface.',
           'reportItemType': 'ERROR'}],
     'MESSAGE': 'Internal Server Error',
     'METHOD': 'POST',
     'REQUEST_PATH': 'https://10.122.197.6:443/rest/interface',
     'RETURN_CODE': 500
}
```